packages/salesforcedx-vscode-core/src/services/sourceStatusSummary.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  Row,
+  Table
+} from '@salesforce/salesforcedx-utils-vscode/out/src/output';
+import { StatusOutputRow } from '@salesforce/source-tracking';
+
+type StatusActualState = 'Deleted' | 'Add' | 'Changed' | 'Unchanged';
+type StatusOrigin = 'Local' | 'Remote';
+
+interface StatusResult {
+  state: string;
+  fullName: string;
+  type: string;
+  filePath?: string;
+  ignored?: boolean;
+  conflict?: boolean;
+  actualState?: StatusActualState;
+  origin: StatusOrigin;
+}
+
+interface FormattedStatusResult {
+  state: string;
+  fullName: string;
+  type: string;
+  filePath?: string;
+  ignored?: string;
+}
+
+export class SourceStatusSummary {
+  constructor(private statusOutputRows: StatusOutputRow[]) {}
+
+  public format(): string {
+    const statusResults = this.statusOutputRows.map(row =>
+      this.resultConverter(row)
+    );
+
+    if (statusResults.length === 0) {
+      return 'No local or remote changes found.';
+    }
+
+    // sort the rows and create a table
+    const sortedStatusResults = statusResults.sort(this.rowSortFunction);
+
+    return new StatusResultsTable(sortedStatusResults).value();
+  }
+
+  /**
+   * STL provides a more useful json output.
+   * This function makes it consistent with the Status command's json.
+   */
+  private resultConverter = (input: StatusOutputRow): StatusResult => {
+    const { fullName, type, ignored, filePath, conflict } = input;
+    const origin = SourceStatusSummary.originMap.get(input.origin) || 'Local';
+    const actualState = SourceStatusSummary.stateMap.get(input.state);
+    return {
+      fullName,
+      type,
+      // this string became the place to store information.
+      // The JSON now breaks out that info but preserves this property for backward compatibility
+      state: `${origin} ${actualState}${conflict ? ' (Conflict)' : ''}`,
+      ignored,
+      filePath,
+      origin,
+      actualState,
+      conflict
```

Comment:
"state" and "conflict" might not be needed here, as I don't think they ever get included in the output to the table when using these 2 commands...